src/Sentry/Laravel/Tracing/Middleware.php

```diff
@@ -91,32 +110,47 @@ private function startTransaction(Request $request, HubInterface $sentry): void
             'url' => '/' . ltrim($request->path(), '/'),
             'method' => strtoupper($request->method()),
         ]);
-        $context->setStartTimestamp($request->server('REQUEST_TIME_FLOAT', $fallbackTime));
+        $context->setStartTimestamp($requestStartTime);
 
         $this->transaction = $sentry->startTransaction($context);
 
         // Setting the Transaction on the Hub
         SentrySdk::getCurrentHub()->setSpan($this->transaction);
 
         if (!$this->addBootTimeSpans() && app() instanceof Laravel) {
-            // @TODO: We might want to move this together with the `RouteMatches` listener to some central place and or do this from the `EventHandler`
-            app()->booted(function () use ($request, $fallbackTime): void {
-                $spanContextStart = new SpanContext();
-                $spanContextStart->setOp('app.bootstrap');
-                $spanContextStart->setStartTimestamp(defined('LARAVEL_START') ? LARAVEL_START : $request->server('REQUEST_TIME_FLOAT', $fallbackTime));
-                $spanContextStart->setEndTimestamp(microtime(true));
-                $this->transaction->startChild($spanContextStart);
-
-                $appContextStart = new SpanContext();
-                $appContextStart->setOp('app.handle');
-                $appContextStart->setStartTimestamp(microtime(true));
-                $this->appSpan = $this->transaction->startChild($appContextStart);
-
-                SentrySdk::getCurrentHub()->setSpan($this->appSpan);
-            });
+            $this->addAppBootstrapSpan($request);
+
+            $appContextStart = new SpanContext();
+            $appContextStart->setOp('app.handle');
+            $appContextStart->setStartTimestamp(microtime(true));
```

Comment:
Okay, how about this, because you make some good points but I feel `microtime(true)` here is still too slow and can be better.

So what if we return the `Span` from the `addAppBootstrapSpan` and IF it's returned (happens because if we don't do the bootstrap span because no booted timestamp or some other reason) we default to `microtime(true)`?

Something like this:

```suggestion
            $appBootedSpan = $this->addAppBootstrapSpan($request);

            $appContextStart = new SpanContext();
            $appContextStart->setOp('app.handle');
            $appContextStart->setStartTimestamp($appBootedSpan ? $appBootedSpan->getStartTimestamp() : microtime(true));
```

And change `addAppBootstrapSpan` to return `Span|null` of course?

Reply:
🤔 this is where I have a question: does `app.handle` includes `app.bootstrap` or should `app.handle` start _after_ the `app.bootstrap`? Because according to your suggestion we would set start timestamp for both `app.handle` and `app.bootstrap` to the same value and _I think_ we actually should set `app.handle` start time to `app.bootstrap` end time - at least based on the original logic that we had in place before I started messing around with it? What do you think?

Reply:
Yes that was what I meant to say 👍  (typo on the start time)